dask/dataframe/core.py

```diff
@@ -7918,7 +7926,13 @@ def repartition(df, divisions=None, force=False):
 def _reduction_chunk(x, aca_chunk=None, **kwargs):
     o = aca_chunk(x, **kwargs)
     # Return a dataframe so that the concatenated version is also a dataframe
-    return o.to_frame().T if is_series_like(o) else o
+    if is_series_like(o):
```

Comment:
What's going on here?

Reply:
This is from #10219.

When you have a DataFrame with ints/floats and bools, the reduction coerces to object, e.g. you get an object dtype series. We transpose to aggregate the chunks, which gives us an all object dtype DataFrame, which isn't what we want. So we have to infer the original dtypes, that's what I added here

Reply:
I see what you're saying, but I'm not completely getting the use case, sorry. Would you have an example?

Reply:
The following example

```
df = pd.DataFrame(
    {
        "a": 1,
        "int": [1, 2],
        "float": [1.0, 2.0],
        "bool": True
    }
)
ddf = dd.from_pandas(df, npartitions=2)

x = ddf.min(numeric_only=True).compute()
```

runs into ``_reduction_chunk`` with
```
a           1
int         1
float     1.0
bool     True
dtype: object
```

Transposing the Series gives all object columns in the DataFrame which would translate to all object columns in the output. ``infer_objects`` casts the dtypes back.

Reply:
👍  thank you, that makes sense.